mesa/visualization/templates/modular_template.html

```diff
@@ -136,15 +141,17 @@
 	// Create the buttons
 	gui.add(this, "reset"); 
 	gui.add(this, "step");
-	gui.add(this, "run");
-	gui.add(this, "pause");
+	var runController = gui.add(this, "run");
 
 	/**
-	Called when the user releases the fps slider; pause the model and rerun, to reset the fps
+	Called when the user releases the fps slider; if already running, stop and
+    start again to reset the fps
 	*/
 	fps_control.onFinishChange(function(value) {
-		pause();
-		run();
+		if (control.running) {
+			run();
+			run();
```

Comment:
The first one stops and deletes the current player, and the second one creates a new one with the new FPS rate. This could be handled more cleanly.